app/assets/javascripts/darkswarm/controllers/order_cycle_controller.js.coffee

```diff
@@ -11,14 +11,15 @@ Darkswarm.controller "OrderCycleCtrl", ($scope, $rootScope, $timeout, OrderCycle
       $("#order_cycle_id").trigger("openTrigger")
 
 
-Darkswarm.controller "OrderCycleChangeCtrl", ($scope, $rootScope, $timeout, OrderCycle, Products, Variants, Cart, ChangeableOrdersAlert) ->
+Darkswarm.controller "OrderCycleChangeCtrl", ($scope, $rootScope, $timeout, $location, OrderCycle, Products, Variants, Cart, ChangeableOrdersAlert) ->
   # Track previous order cycle id for use with revertOrderCycle()
   $scope.previous_order_cycle_id = OrderCycle.order_cycle.order_cycle_id
   $scope.$watch 'order_cycle.order_cycle_id', (newValue, oldValue)->
     $scope.previous_order_cycle_id = oldValue
 
   $scope.changeOrderCycle = ->
     OrderCycle.push_order_cycle $scope.orderCycleChanged
+    $location.path("shop")
```

Comment:
Spree does it with app/views/spree/admin/shared/_routes.html.erb
I dont think we should do routing in angular. The main reason is that we don't really need it and it can get complicated if we go for it.
In this case, I'd do this on the server. For example, making push_order_cycle return a 301 and follow it on the callback?

Reply:
This changes the hash only, that's done client side no?

Reply:
I think in this case we already are handling it in Angular, and we update the browser path when different tabs are clicked...

Reply:
@mbudm it's a call to the server, see app/assets/javascripts/darkswarm/services/order_cycle.js.coffee
that goes to app/controllers/shop_controller.rb#order_cycle

Reply:
>> we already are handling it in Angular

Matt, where? I think the actual route is on the server side, isnt it?

Reply:
@luisramos0 the shop is essentially a SPA. Changing tabs is done with an anchor URL. See: https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/shopping_shared/_tabs.html.haml#L12

So changing the hash via javascript is the same as a click on `<a href="#/shop">`.

The network traffic shows this when changing tabs - it's all XHR (and one PNG). This network traffic is initiated from javascript by the OrderCycleCtrl when the shop tab is rendered. Those calls every time are probably not necessary most of the time as I doubt shops product change that quickly but thats a future optimisation.
![ofn-shop-network-calls-720](https://user-images.githubusercontent.com/2467577/87385390-4722d500-c5e1-11ea-820c-6aa79be3de22.gif)


What I did find in my travels is that in the admin we are using a [tab directive,](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/assets/javascripts/darkswarm/directives/tabset_ctrl.js.coffee#L18) that has a method for selecting tabs, and broadcasts this change to any listeners. Perhaps this should also be used in the shop as it's nicer than changing the location directly. As an aside this directive uses the same `$location.path('tabname')` approach (as do a few other examples).

Reply:
Hey Steve, that tabset directive is used in the frontoffice, it's used in the tabs on the user account page.

That tbaset code is a good example where I dont think we should go. Even on this shopfront page, I think it's best to avoid SPA altogether and keep with rails routes. Simple, it works, we dont need the extra complication.

But I understand that the current routing is done with through the fragment/hash already. Because of that I dont think this is a major problem. I think the routing needs to be done at least in an isolated manner though, for example, in a specific service? Having the OrderCyycleChangeController know about routes looks wrong.

Not to mention the fact that we are talking about an angular file with two controllers defined in it...

I am just sharing my concerns and why I think we should make it a little better as we change it.

Reply:
OK I found a nicer way - using the existing broadcast event of an orderCycleChange, which is picked by by the shop tab controller - generically called `PageSelectionCtrl` (rename?). To make this work I also removed what seems to be a redundant piece of code that fires this event whenever the OrderCycleCtrl is instantiated.

I suspect that this was a way to open the select box (clue is in the "openTrigger" trigger argument) to prompt users to select an order, which has been superceded by the red style applied to the order cycle selector:

<img width="916" alt="Screen Shot 2020-07-15 at 3 36 48 pm" src="https://user-images.githubusercontent.com/2467577/87508919-c5e84280-c6b3-11ea-9a69-68976bdd2e7f.png">

The commits around this area in git blame seem to cover functionality that still works with this code removed:
<img width="1125" alt="Screen Shot 2020-07-15 at 3 59 06 pm" src="https://user-images.githubusercontent.com/2467577/87509135-47d86b80-c6b4-11ea-819f-abda7406aca6.png">

There is defs a lot more refactoring potential in the shop, and probably a lot more redundant code that is hard to spot at first - it is also high value given this is the crucial customer facing experience. Possibly a candidate if we look at a library upgrade/change.